src/acl_auto_configure.cpp

```diff
@@ -706,13 +705,16 @@ bool acl_load_device_def_from_str(const std::string &config_str,
 
         std::string buffer_location = "";
         if (result) {
+          unsigned int num_buffer_locations = 0;
           result = result && read_uint_counters(config_str, curr_pos,
                                                 num_buffer_locations, counters);
-          for (auto k = 0U; result && (k < num_buffer_locations); k++) {
-            if (j > 0) {
-              buffer_location = buffer_location + " ";
-            }
-
+          if (result && num_buffer_locations > 1) {
+            std::cerr
+                << "WARNING: kernel argument has multiple buffer_location "
+                   "attributes which is not supported.\nSelecting last "
+                   "buffer_location attribute.\n";
```

Comment:
I think it might be nicer to let user know which is the last one, I am not sure if the order of the buffer location is the same as how they passed it into the property list.

e.g. a format string may look like: `("Warning: kernel argument has multiple buffer_location attributes which is not supported. Selecting %s as buffer location for kernel argument %d", buffer_location, j)`

Reply:
Thanks Sherry, that's a good suggestion. The attribute value is easy to print, however printing the argument index is slightly trickier since `j` is no longer available when this code is factored out (which uncovered that this code was buggy in the first place). A proper solution would be to have hierarchical error handling, i.e. the outer function prints "error while parsing kernel argument %d" and the inner function prints "selecting %s as buffer location". However, since this code is currently not unit-tested, I would like to keep refactoring to a minimum and only print the attribute value for now.

Reply:
I have updated the message to include `buffer_location`. As mentioned above, including the kernel argument index would be useful to the user, but it will have to wait for later refactoring, after increasing code coverage.